src/hooks/useEvents.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from "react";
+import { cloneObject } from "../utils";
+
+export function useEvents() {
+  const [events, setEvents] = useState<Array<MainEvent>>([]);
+
+  useEffect(() => {
+    const savedEvents = localStorage.getItem("events");
+    if (savedEvents) {
+      setEvents(JSON.parse(savedEvents));
+    }
+  }, []);
+
+  return { events, addEvent, updateEvent, deleteEvent };
+
+  function addEvent(eventData: MainEvent) {
+    const newEvent = {
+      ...eventData,
+      id: Date.now(),
+      attendeeList: [],
+    };
+
+    const newEvents = cloneObject(events) as Array<MainEvent>;
+    newEvents.push(newEvent);
+
+    setEvents(newEvents);
+    localStorage.setItem("events", JSON.stringify(newEvents));
+  }
+
+  function updateEvent(eventData: MainEvent) {
+    const newEvents = (cloneObject(events) as Array<MainEvent>).map((event) =>
```

Comment:
why using `as` here?

Reply:
The `cloneObject` function takes an `Object` or an `Array` and returns a deep copy of the `Object` or `Array` and the returned `Array` or `Object` are plain javascript `Objects` that is why the type assertion `as` was needed here to make typescript understand that the returned `Object` is a `Array<MainEvent>` type